evm_loader/program/src/instruction/transaction_step.rs

```diff
@@ -117,15 +117,27 @@ fn allocate_or_reinit_state(
     if is_allocate {
         storage.reset_steps_executed();
 
-        // Dealloc objects that were potentially alloced in previous iterations before the reset.
+        // Dealloc evm that was potentially alloced in previous iterations before the reset.
         if storage.is_evm_alloced() {
             storage.dealloc_evm::<EvmBackend, NoopEventListener>();
         }
+
+        // Dealloc executor state that was potentially alloced in previous iterations before the reset.
+        // Also, copy the cache for use into the new ExecutorStateData.
+        let mut cache = None;
```

Comment:
The cache may contain other data in the future. But here we only need to restore the timestamp and the block.

Reply:
done - right now `BlockParams` is separate from `Cache` so we won't have such a problem in the future.